### ISSUE TITLE
Blank screen when launching external apps from CHT Android app

### ISSUE THREAD

**Issue by IreneAchola** [medic/cht-core#6648]
**Describe the bug**
Opening links to external applications does not work when launched from the android application (no application selector) but works when launched from the browser.

**To Reproduce**

1. Install CHT CHA Learning Care application from [here](https://drive.google.com/file/d/1CicO_voKTYv6NctrWfEAzXO_hVeAHEq3/view)
2. Login as CHW
3. Select any task in the tasks tab
4. Click **Open Oppia Mobile** button
5. See a blank white screen

**Expected behavior**
User should be able to see an application popup to select which application to open the link with

**Screenshots**

Application screenshots

![20200921_154909](https://user-images.githubusercontent.com/8436873/93768565-3feff700-fc22-11ea-93ef-09554c58e5c7.jpg)

Browser screenshots

![20200921_154657](https://user-images.githubusercontent.com/8436873/93768559-3d8d9d00-fc22-11ea-9de7-a51da089aa95.jpg)

**Environment**
- Instance: cht-covid.dev.medicmobile.org
- Browser: Worked in both Firefox and Chrome
- Client platform: Android OS, Linux
- App: webapp
- Version: 3.10.0, observed in previous version as well (CIHA Rapid Pro version)

**Additional context**
Following steps 2 to 4 using the browsers on the phone prompts user to select an application to open the link with

**Comment by garethbowen** [comment]
@IreneAchola Can you share the configuration you're using?

**Comment by IreneAchola** [comment]
@garethbowen Config available on [this branch](https://github.com/medic/config-covid/tree/cht-oppia-mvp/cht_oppia_mvp)

**Comment by dianabarsan** [comment]
This might be relevant: https://github.com/medic/medic-android/blob/master/src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java#L383

We're hooking into a crosswalk feature that intercepts page loads and we block loading external pages.

**Comment by dianabarsan** [comment]
Yup it looks like that's the case: 
https://github.com/crosswalk-project/crosswalk/issues/4000 <- see the solution provided here: 

```
xDemo.setResourceClient(new XWalkResourceClient(xDemo) {
    @Override
    public boolean shouldOverrideUrlLoading(XWalkView view, String url) {
        view.loadUrl(url);
        return true;
    }
}); 
```

We're blocking activities for any URL that is not a `tel` or `sms` https://github.com/medic/medic-android/blob/master/src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java#L371

**Comment by dianabarsan** [comment]
I think the correct solution would be to change this code to launch activities for external URLs as well, and use the same URL matching as we do in the other function.

**Comment by garethbowen** [comment]
Thanks for confirming Diana. I've put this in 3.11.0 as it's blocking the Oppia integration project.

**Comment by isaacholeman** [comment]
Great to see this investigated and prioritized so quickly, thank you!

**Comment by garethbowen** [comment]
Ready for AT in `6648-enable-outside-links`

**Comment by dianabarsan** [comment]
Just a small addendum that this is a `medic-android` issue, the branch above is in `medic-android`: https://github.com/medic/medic-android/tree/6648-enable-outside-links

To test, you'd need to add some config (a form maybe) that has an external link that you can click in the android app. 
Also, please make sure that the "call" button from the contact profile still works.

**Comment by newtewt** [comment]
I think this is ready to go @dianabarsan. 

Tested with the CHA app not installed. This navigates us to the webpage.
Tested with the CHA app installed and used the form to navigate to the app. 
Navigates back to our app as expected. 
Tested the phone with no internet and navigate to the CHA app. 
I was able to use the call function and get to the dialer app. I don't know if we have a way to go directly to the messaging app installed so I couldn't test that.
